Podverse/PlaylistsTableViewController.swift

```diff
@@ -22,76 +22,115 @@ class PlaylistsTableViewController: PVViewController {
         
         PVAuth.shared.delegate = self
         
-        loadPlaylistData()
+        retrievePlaylists()
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        checkAuthorizationAndConnectivity()
     }
     
-    func checkAuthorizationAndConnectivity() {
-        var message = ErrorMessages.noPlaylistsAvailable.text
-        var buttonTitle = "Retry"
-        var selector:Selector = #selector(PlaylistsTableViewController.loadPlaylistData)
+    func retrievePlaylists() {
+        
+        guard checkForConnectivity(), checkForAuthorization() else {
+            return
+        }
+        
+        self.hideNoDataView()
+        self.activityIndicator.startAnimating()
+        
+        Playlist.retrievePlaylistsFromServer() { (playlists) -> Void in
+            self.reloadPlaylistData(playlists: playlists)
+        }
+    }
+    
+    func checkForAuthorization() -> Bool {
+        
+        var message = Strings.Errors.noPlaylistsNotLoggedIn
+        var buttonTitle = "Login"
+        let selector = #selector(PlaylistsTableViewController.presentLogin)
+        
         let isLoggedIn = PVAuth.userIsLoggedIn
         
-        if self.reachability.hasInternetConnection() == false {
-            message = ErrorMessages.noPlaylistsInternet.text
+        guard PVAuth.userIsLoggedIn else {
+            loadNoDataView(message: message, buttonTitle: buttonTitle, buttonPressed: selector)
+            return false
         }
-        else if !isLoggedIn {
-            message = ErrorMessages.noPlaylistsNotLoggedIn.text
-            buttonTitle = "Login"
-
-            selector = #selector(PlaylistsTableViewController.presentLogin)
+        
+        return true
+        
+    }
+    
+    func checkForConnectivity() -> Bool {
+        var message = Strings.Errors.noPlaylistsInternet
+        var buttonTitle = "Retry"
+        var selector:Selector = #selector(PlaylistsTableViewController.retrievePlaylists)
+        
+        guard self.reachability.hasInternetConnection() else {
```

Comment:
Seems that reachability is not needed here since you already have the global checkForConnectivity function. You can probably just use that one.